test/integration/secret_test.go

```diff
@@ -0,0 +1,185 @@
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/master"
+	"github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/admission/admit"
+)
+
+func init() {
+	requireEtcd()
+}
+
+func deletePodOrErrorf(t *testing.T, c *client.Client, ns, name string) {
+	if err := c.Pods(ns).Delete(name); err != nil {
+		t.Errorf("unable to delete pods %v: %v", name, err)
+	}
+}
+func deleteSecretOrErrorf(t *testing.T, c *client.Client, ns, name string) {
+	if err := c.Secrets(ns).Delete(name); err != nil {
+		t.Errorf("unable to delete secret %v: %v", name, err)
+	}
+}
+
+// TestSecrets tests apiserver-side behavior of creation of secret objects and their use by pods.
+func TestSecrets(t *testing.T) {
+	helper, err := master.NewEtcdHelper(newEtcdClient(), "v1beta1")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	var m *master.Master
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		m.Handler.ServeHTTP(w, req)
+	}))
+	defer s.Close()
+
+	m = master.New(&master.Config{
+		EtcdHelper:        helper,
+		KubeletClient:     client.FakeKubeletClient{},
+		EnableLogsSupport: false,
+		EnableUISupport:   false,
+		EnableIndex:       true,
+		APIPrefix:         "/api",
+		Authorizer:        apiserver.NewAlwaysAllowAuthorizer(),
+		AdmissionControl:  admit.NewAlwaysAdmit(),
+	})
+
+	testCases := []string{
+		"v1beta1",
+		"v1beta2",
+	}
+
+	for _, apiVersion := range testCases {
+		deleteAllEtcdKeys()
+		client := client.NewOrDie(&client.Config{Host: s.URL, Version: apiVersion})
+		DoTestSecrets(t, client, apiVersion)
+	}
+}
+
+// DoTestSecrets test secrets for one api version.
+func DoTestSecrets(t *testing.T, client *client.Client, apiVersion string) {
+	// Make a secret object in each of two namespaces.
+	ns1 := "ns1"
+	ns2 := "ns2"
+	secrets := []api.Secret{
+		{
+			ObjectMeta: api.ObjectMeta{
+				Name:      "secret",
+				Namespace: ns1,
+			},
+			Data: map[string][]byte{
+				"data": []byte("value1\n"),
+			},
+		},
+		{
+			ObjectMeta: api.ObjectMeta{
+				Name:      "secret",
+				Namespace: ns2,
+			},
+			Data: map[string][]byte{
+				"data": []byte("value2\n"),
+			},
+		},
+	}
+
+	for _, s := range secrets {
+		defer deleteSecretOrErrorf(t, client, s.Namespace, s.Name)
+		if _, err := client.Secrets(s.Namespace).Create(&s); err != nil {
+			t.Errorf("unable to create test secret %s: %v", s.Name, err)
+		}
+	}
+
+	// Template for pods that use a secret.
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Name: "XXX",
+		},
+		Spec: api.PodSpec{
+			Volumes: []api.Volume{
+				{
+					Name: "secvol",
+					VolumeSource: api.VolumeSource{
+						Secret: &api.SecretVolumeSource{
+							Target: api.ObjectReference{
+								Kind:      "Secret",
+								Namespace: "XXX",
```

Comment:
I assume that this "XXX" literal and the one above are not required to be equal?  If so, suggest making them non-equal to clarify that there's no relationship between the two.